dbt/adapters/athena/impl.py

```diff
@@ -694,48 +694,47 @@ def list_relations_without_caching(self, schema_relation: AthenaRelation) -> Lis
                 region_name=client.region_name,
                 config=get_boto3_config(num_retries=creds.effective_num_retries),
             )
-        paginator = glue_client.get_paginator("get_tables")
-
-        kwargs = {
-            "DatabaseName": schema_relation.schema,
-        }
-        # If the catalog is `awsdatacatalog` we don't need to pass CatalogId as boto3 infers it from the account Id.
-        if catalog_id := get_catalog_id(data_catalog):
-            kwargs["CatalogId"] = catalog_id
-        page_iterator = paginator.paginate(**kwargs)
-
-        relations = []
-        quote_policy = {"database": True, "schema": True, "identifier": True}
 
+        catalog_id = get_catalog_id(data_catalog)
+        paginator = glue_client.get_paginator("get_tables")
         try:
-            for page in page_iterator:
-                tables = page["TableList"]
-                for table in tables:
-                    if "TableType" not in table:
-                        LOGGER.debug(f"Table '{table['Name']}' has no TableType attribute - Ignoring")
-                        continue
-                    _type = table["TableType"]
-                    _detailed_table_type = table.get("Parameters", {}).get("table_type", "")
-                    if _type == "VIRTUAL_VIEW":
-                        _type = self.Relation.View
-                    else:
-                        _type = self.Relation.Table
-
-                    relations.append(
-                        self.Relation.create(
-                            schema=schema_relation.schema,
-                            database=schema_relation.database,
-                            identifier=table["Name"],
-                            quote_policy=quote_policy,
-                            type=_type,
-                            detailed_table_type=_detailed_table_type,
-                        )
-                    )
+            tables = (
+                paginator.paginate(DatabaseName=schema_relation.schema, CatalogId=catalog_id)
```

Comment:
if the catalog_id is None, we don't have to pass it.
I think that we can still keep:
```
kwargs = {
            "DatabaseName": schema_relation.schema,
        }
if catalog_id := get_catalog_id(data_catalog):
            kwargs["CatalogId"] = catalog_id
```

then call the paginator with the expended kwargs

Reply:
fixed by
[2a2be56](https://github.com/dbt-athena/dbt-athena/pull/629/commits/2a2be56e57e0a67b391d6f61ef1bf2b7fa18a0b6)